### PR TITLE
onagre: wrap with x11 libraries

### PR DESCRIPTION
**Pull request by Septias**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/onagre/default.nix [hunk 1]
```diff
@@ -5,6 +5,8 @@
 , pkgconf
 , freetype
 , expat
+, xorg
```
Comment (AndersonTorres) [inline pkgs/applications/misc/onagre/default.nix:8]: Do not use xorg indirection.

```suggestion
, libX11
, libXcursor
, libXi
, libXrandr
```

File: pkgs/applications/misc/onagre/default.nix [hunk 2]
```diff
@@ -30,6 +32,11 @@ rustPlatform.buildRustPackage rec {
   nativeBuildInputs = [ cmake pkgconf ];
   buildInputs = [ freetype expat ];
 
+  libs = with xorg; pkgs.lib.makeLibraryPath [libX11 libXcursor libXrandr libXi];
+  postInstall = ''
+    wrapProgram $out/bin/onagre --prefix LD_LIBRARY_PATH : $libs --prefix PATH : ${pop-launcher}/bin
+  '';
+
```
Comment (AndersonTorres) [inline pkgs/applications/misc/onagre/default.nix:39]: **Never** use `pkgs.`.

```suggestion
  postInstall = ''
    wrapProgram $out/bin/onagre \
      --prefix LD_LIBRARY_PATH ":" ${lib.makeLibraryPath [ libX11 libXcursor libXi libXrandr ];} \
      --prefix PATH ":" ${lib.getExe pop-launcher}
  '';

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Reword the pull request according to contributing guide.

**Comment by AndersonTorres** [comment]
Squash the commits into only one, according to the contributing guide.

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 290409` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>onagre</li>
  </ul>
</details>

**Comment by Septias** [comment]
Testing with `nixpkgs-review pr 290409` times somehow...?

**Comment by AndersonTorres** [comment]
Have you tested locally?
I can try it later when I'm home.

Besides, Lgtm.

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 290409` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>onagre</li>
  </ul>
</details>

**Comment by Septias** [comment]
I updated to v1.0.0 and it now clearly states, that pop-launcher is not available. Why did you wrap `:` in `"`? @AndersonTorres

File: pkgs/applications/misc/onagre/default.nix [hunk 3]
```diff
@@ -5,31 +5,40 @@
...
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "onagre";
-  version = "1.0.0-alpha.0";
+  version = "1.0.0";
 
   src = fetchFromGitHub {
     owner = "oknozor";
     repo = pname;
```
Comment (AndersonTorres) [inline pkgs/applications/misc/onagre/default.nix:22]:
```suggestion
    repo = "onagre";
```

https://github.com/NixOS/nixpkgs/issues/277994

**Review by AndersonTorres — COMMENTED** [review]
1. Squash the commits
2. Reword the commit as `onagre: <oldver> -> <newver>`

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 290409` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>onagre</li>
  </ul>
</details>

**Comment by jfvillablanca** [comment]
> Result of `nixpkgs-review pr 290409` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package built:

It builds and runs. I ran `./results/onagre/bin/onagre` and the menu pops out. However, when I try to run it again (to take a screenshot to show that it works), it automatically selects (and runs) the first result and exits. Is anyone else experiencing this?

**Comment by Septias** [comment]
Had the same problem from time to time @jfvillablanca 
Seems to be an onagre problem

File: pkgs/applications/misc/onagre/default.nix [hunk 4]
```diff
@@ -5,31 +5,40 @@
...
   cargoSha256 = "sha256-IOhAGrAiT2mnScNP7k7XK9CETUr6BjGdQVdEUvTYQT4=";
 
-  nativeBuildInputs = [ cmake pkgconf ];
+  nativeBuildInputs = [ cmake pkgconf makeWrapper ];
   buildInputs = [ freetype expat ];
 
+  postInstall = ''
+    wrapProgram $out/bin/onagre \
+      --prefix LD_LIBRARY_PATH ":" ${lib.makeLibraryPath [ libX11 libXcursor libXi libXrandr ]} \
+      --prefix PATH : ${pop-launcher}/bin
```
Comment (AndersonTorres) [inline pkgs/applications/misc/onagre/default.nix:39]:
```suggestion
      --prefix PATH : ${lib.getBin pop-launcher}/bin
```

Better safe than sorry.

**Review by AndersonTorres — APPROVED** [review]
Besides, LGTM

File: pkgs/applications/misc/onagre/default.nix (hunk 4 above)
Comment (Septias) [reply pkgs/applications/misc/onagre/default.nix:39]: I'm pretty sure I tried that and it didn't work :thinking:

**Comment by jfvillablanca** [comment]
> Had the same problem from time to time @jfvillablanca Seems to be an onagre problem

Can you please confirm if it is an upstream issue 
If it's not, we can have this PR merged. Thank you :smile:

File: pkgs/applications/misc/onagre/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/applications/misc/onagre/default.nix:39]: Therefore it deserves a fixup on pop-launcher!

But let's ignore this for now.

**Pull request closed without merging**